src/pybind/mgr/heartbeat_webhook/module.py

```diff
@@ -0,0 +1,126 @@
+import logging
+import json
+from mgr_module import MgrModule, Option, HandleCommandResult, CLIReadCommand
+from typing import Any, List, TYPE_CHECKING
+from threading import Event
+import urllib3
+
+
+class HeartbeatWebhook(MgrModule):
+    MODULE_OPTIONS = [
+        Option(
+            name='interval',
+            type='secs',
+            default=60,
+            desc='How frequently to reexamine health status',
+            runtime=True),
+        Option(
+            name='webhook_url',
+            type='string',
+            default='',
+            desc='Url to send webhooks to',
+            runtime=True),
+        Option(
+            name='send_on_warn',
+            type='bool',
+            default=False,
+            desc="Continue sending webhooks if the cluster health is HEALTH_WARN")
+    ]
+    NATIVE_OPTIONS: List[str] = [
+    ]
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super(HeartbeatWebhook, self).__init__(*args, **kwargs)
+
+        self.run = True
+        self.event = Event()
+        
+        self.config_notify()
+
+        self.log.info("Initializing heartbeat_webhook module")
+
+        self.http = urllib3.PoolManager()
+
+        if TYPE_CHECKING:
```

Comment:
I think when I originally wrote this I had assumed it was convention since the examples I was looking at had it, but looking at it again it does seem unnecessary. I'll go ahead and remove it.